src/Command/SetSettingsCommand.php

```diff
@@ -0,0 +1,182 @@
+
+    private const ARGUMENT_TYPE = 'type';
+
+    private const ARGUMENT_VALUE = 'value';
+
+    private RegistryInterface $settingsRegistry;
+
+    private ChannelRepositoryInterface $channelRepository;
+
+    private FactoryInterface $settingFactory;
+
+    private SettingRepositoryInterface $settingRepository;
+
+    private EntityManagerInterface $settingManager;
+
+    private SettingsFormatterInterface $settingsFormatter;
+
+    protected static $defaultName = 'monsieurbiz:settings:set';
+
+    public function __construct(
+        RegistryInterface $settingsRegistry,
+        ChannelRepositoryInterface $channelRepository,
+        FactoryInterface $settingFactory,
+        SettingRepositoryInterface $settingRepository,
+        EntityManagerInterface $settingManager,
+        SettingsFormatterInterface $settingsFormatter,
+        string $name = null
+    ) {
+        $this->settingsRegistry = $settingsRegistry;
+        $this->channelRepository = $channelRepository;
+        $this->settingFactory = $settingFactory;
+        $this->settingRepository = $settingRepository;
+        $this->settingManager = $settingManager;
+        $this->settingsFormatter = $settingsFormatter;
+        parent::__construct($name);
+    }
+
+    protected function configure(): void
+    {
+        $this
+            ->setDescription('Set a settings value for a given path')
+            ->setHelp('This command allows you to set a settings value for a given path')
+            ->addArgument(self::ARGUMENT_ALIAS, InputArgument::REQUIRED, 'Alias of the settings like {vendor}.{plugin} from the setting definition')
+            ->addArgument(self::ARGUMENT_PATH, InputArgument::REQUIRED, 'Path of the settings')
+            ->addArgument(self::ARGUMENT_TYPE, InputArgument::REQUIRED, 'Type of the settings', null)
+            ->addArgument(self::ARGUMENT_VALUE, InputArgument::REQUIRED, 'Value of the settings')
+            ->addOption(self::OPTION_CHANNEL, 'c', InputOption::VALUE_OPTIONAL, 'Channel code')
+            ->addOption(self::OPTION_LOCALE, 'l', InputOption::VALUE_OPTIONAL, 'Locale code')
+        ;
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output): int
+    {
+        try {
+            /** @var string $alias */
+            $alias = $input->getArgument(self::ARGUMENT_ALIAS);
+            /** @var string $path */
+            $path = $input->getArgument(self::ARGUMENT_PATH);
+            $channelCode = $input->getOption(self::OPTION_CHANNEL);
+            /** @var ?string $locale */
+            $locale = $input->getOption(self::OPTION_LOCALE);
+
+            $channel = null;
+            if (null !== $channelCode) {
+                /** @var ?ChannelInterface $channel */
+                $channel = $this->channelRepository->findOneBy(['code' => $channelCode]);
+            }
+
+            /** @var SettingsInterface $settings */
+            $settings = $this->settingsRegistry->getByAlias($alias);
```

Comment:
**🎯 suggestion:** ‏Can be null right? So maybe we could add a test here and throw an error if we don't find the Settings by its alias.